x/allocation/keeper/keeper.go

```diff
@@ -0,0 +1,157 @@
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// fund validator rewards pool
+	validatorRewardsCoins := k.GetProportions(ctx, blockInflation, proportions.ValidatorRewards)
+	if !validatorRewardsCoins.IsZero() {
+		err := k.DistributeValidatorRewards(ctx, validatorRewardsCoins)
+		if err != nil {
+			return err
+		}
+	}
+	devRewards := k.GetProportions(ctx, blockInflation, proportions.DeveloperRewards)
+	err := k.DistributeDeveloperRewards(ctx, blockInflationAddr, devRewards, params.WeightedDeveloperRewardsReceivers)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k Keeper) DistributeDeveloperRewards(ctx sdk.Context, feeCollectorAddress sdk.AccAddress, devRewards sdk.Coin, devs []types.WeightedAddress) error {
+	for _, w := range devs {
+		devRewardPortionCoins := sdk.NewCoins(k.GetProportions(ctx, devRewards, w.Weight))
+		if w.Address != "" {
+			devRewardsAddr, err := sdk.AccAddressFromBech32(w.Address)
+			if err != nil {
+				return err
+			}
+			err = k.bankKeeper.SendCoins(ctx, feeCollectorAddress, devRewardsAddr, devRewardPortionCoins)
+			if err != nil {
+				return err
+			}
+			k.Logger(ctx).Debug("sent coins to developer", "amount", devRewardPortionCoins.String(), "from", feeCollectorAddress)
+		}
+	}
+	return nil
+}
+
+// DistributeValidatorRewards distributes rewards to validators
+func (k Keeper) DistributeValidatorRewards(ctx sdk.Context, rewards sdk.Coin) error {
+	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, authtypes.FeeCollectorName, types.ValidatorRewardsPool, sdk.NewCoins(rewards))
+	if err != nil {
+		return err
+	}
+	validators := k.stakingKeeper.GetLastValidators(ctx)
+	// this should never happen but adding the validation
+	if len(validators) == 0 {
+		return nil
+	}
+	// get the amount of coins to distribute to each validator
+	validatorReward := rewards.Amount.QuoRaw(int64(len(validators)))
+	// distribute coins to each validator by accumulating their rewards
+	// the module account will hold the tokens until they are withdrawn by validators
+	for _, v := range validators {
+		// get the validator operator address
+		operator, err := sdk.ValAddressFromBech32(v.OperatorAddress)
+		// error should never happen as stored validator addresses must always be valid
+		if err != nil {
+			return err
+		}
+		// we just need to cast directly to sdk.AccAddress because the bech32 parsing
+		// was previously validated and the underlying bytes are the same
+		accAddr := sdk.AccAddress(operator.Bytes())
+		r := k.GetValidatorRewards(ctx, accAddr)
+		if r.Rewards != nil && !r.Rewards.Empty() && !r.Rewards.IsZero() {
+			// add to existing rewards
+			r.Rewards = r.Rewards.Add(sdk.NewCoin(rewards.Denom, validatorReward))
+		} else {
+			// initialize rewards
+			r = types.ValidatorReward{
+				Address: accAddr.String(),
+				Rewards: sdk.NewCoins(sdk.NewCoin(rewards.Denom, validatorReward)),
+			}
+		}
+		k.SetValidatorRewards(ctx, accAddr, r)
```

Comment:
The types.ValidatorReward type is pretty big, including the address which is already in the key. If we want to avoid doing a bank send here for efficiency, doesn't it make sense to store the reward as a plain uint64 value (8 bytes)?

Reply:
stored as uint